lib/src/repo_path.rs

```diff
@@ -814,6 +825,17 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_extend() {
+        let mut path = RepoPath::root().to_owned();
+        path.extend([repo_path_component("dir")]);
+        assert_eq!(path.as_ref(), repo_path("dir"));
+        path.extend(std::iter::repeat_n(repo_path_component("subdir"), 3));
+        assert_eq!(path.as_ref(), repo_path("dir/subdir/subdir/subdir"));
+        path.extend(std::iter::empty::<RepoPathComponentBuf>());
```

Comment:
nit: Also test this on when the path is empty. I could imagine a bug causing it to make the path change from "" to "/" (it would happen if you moved the condition to be before the loop).

Reply:
`RepoPathComponent[Buf]` enforce that a path component cannot be empty. I _could_ use `RepoPathComponent::new_unchecked("")` in the test but I see a point in testing behavior for unsound input.

That being said, shouldn't `new_unchecked()` or `from_internal_string_unchecked()` be `unsafe`?